metaflow/plugins/datatools/s3/s3.py

```diff
@@ -250,8 +251,25 @@ def blob(self):
         bytes
             Contents of the object as bytes.
         """
+        _CHUNKSIZE = 8192
+
+        def _read_file_py35(file):
```

Comment:
This function also feels very inefficient. We had to do something similar in read_in_chunks (which for some reason is present in two places so you can also remove one if you want). You could use a similar approach using a BytesIO or something like that instead of creating an array of bytes and then joining them. It may be a bit more efficient.